src/patient-registration/patient-registration.component.tsx

```diff
@@ -75,7 +82,19 @@ export const PatientRegistration: React.FC = () => {
     );
     return () => abortController.abort();
   }, []);
+  const config = useConfig();
+  const [personAttConfig, setPersonAttConfig] = useState([]);
 
+  useEffect(() => {
+    config.personAttributes.map((c, i) => {
+      const atts = {
+        label: c.label,
+        uuid: c.uuid,
+        placeholder: c.placeholder,
+      };
+      setPersonAttConfig(personatt => [...personatt, atts]);
```

Comment:
The attribute definitions are input, not state. You shouldn't be changing them. So don't define `personAttConfig`. Just use `config.personAttributes`.

Reply:
(i.e., I don't think any of this code is necessary)

Reply:
@brandon The object array added to the patient object [here](https://github.com/openmrs/openmrs-esm-patient-registration/pull/22/files#diff-42492693143d921dd7a3dafa539ba188R120) is expected to be having two properties as seen on [line 103 and 104](https://github.com/openmrs/openmrs-esm-patient-registration/pull/22/files#diff-42492693143d921dd7a3dafa539ba188R119-R120). However our config comes with a couple of metadata that we need like [placeholders](https://github.com/openmrs/openmrs-esm-patient-registration/pull/22/files#diff-42492693143d921dd7a3dafa539ba188R91-R93) which we dont need when saving the attributes thus I decided to create a state to hold all the configs before passing only the two required properties to the [form here](https://github.com/openmrs/openmrs-esm-patient-registration/pull/22/files#diff-42492693143d921dd7a3dafa539ba188R170)

Reply:
Yeah, there's no reason to do that.

Reply:
@brandones  am lost....Could you give me a sketch of what the code shud look like?